### DISCUSSION TITLE
Feature request for additional fields for CVS export

### DISCUSSION THREAD

**Discussion by PerpSearch** [benjaminjonard/koillection#1109] [Ideas]
First, thanks. Koillection is working very well in helping me manage my collection. It seems extremely flexible and pretty fast in how it handles searches.  I cannot imagine the time you have invested in keeping this up, but I'm assuming it's a labor of love that meats your own needs as well.  

As you tweak, revise, and improve Koillection there are a couple things I like to see eventually.  I would find it very useful to have some extra options under the 'Tools' menu. At present I can only export "printable lists" and "CVS" files and that only includes a check box and the item name. 
  
I believe it would be useful to be able to select additional fields from the item record to export in CVS format as well.  I have a collection of 232 antique photographs. A record would consist of the Name and a Template comprised of 1 image and 15 other data fields.
 Looks like this: 

![2024-04-29_155941](https://github.com/benjaminjonard/koillection/assets/92053831/df61de99-8d37-4a94-a182-787db6919d91)

The container is 1.85 GB with the DB is at 68.8 MB.    

It's easy to ask as I don't have a clue how much coding goes into it, but I also think this is could be part of Koillection's natural evolution anyway and I'm just encouraging you to do something you're already planning.  Either way I look forward to any refinements you've planned.  Again, thanks for your work on this.

**Comment by benjaminjonard** [comment]
Hello, 

The current CSV export is a very old feature I've never used myself. I think it needs to be revamped to be really useful.

Currently, I'm considering adding a new page where you can select the collections you want to export, along with additional fields for each collection.

I'll take some time to think about how to tackle this.

**Reply by PerpSearch** [reply]
Thanks for considering it.  Personally, it would solve a couple of issues for me.  I like to be able to create reports and lists, but we don't have that.  

An example for me would be if I wanted a list of all the items that I purchased last year over $30.  There's really no easy way to get that.  

A built in reporting option with selected fields and filtering would do it, but with a CVS export I can dump it into a spreadsheet and filter and sort whatever I want and it's probably less coding on your end.  

That was my thinking anyway.  The program is good, I like it, and if you get around to adding this feature (or any for that matter) I am grateful.

Thanks again,
C
